pkg/handlers/s3_handler.go

```diff
@@ -191,9 +191,12 @@ func (h *s3Handler) OnDelete(ns string, vc kvcv1.VolumeConfig, controllerRef met
 
 	for _, resource := range podList {
 		resControllerRef := metav1.GetControllerOf(resource)
+		if resControllerRef == nil {
```

Comment:
Same comment here as above. Maybe a TODO would help.

Reply:
No we should not delete these resources. They should not manipulated. This case is true when a resource has the same labels as a volume manager CR.